src/taipy/core/_repository/_sql_repository.py

```diff
@@ -45,56 +45,79 @@ def __init__(self, model_type: Type[ModelType], converter: Type[Converter], sess
     ###############################
     def _save(self, entity: Entity):
         obj = self.converter._entity_to_model(entity)
-        if self.db.query(self.model_type).filter_by(id=obj.id).first():
-            self.__update_entry(obj)
+        if self._exists(entity.id):  # type: ignore
+            self._update_entry(obj)
             return
         self.__insert_model(obj)
 
     def _exists(self, entity_id: str):
-        return bool(self.db.query(self.model_type.id).filter_by(id=entity_id).first())  # type: ignore
+        query = self.model_type.__table__.select().filter_by(id=entity_id)  # type: ignore
+        return bool(self.db.execute(str(query), [entity_id]).fetchone())
 
     def _load(self, entity_id: str) -> Entity:
-        if entry := self.db.query(self.model_type).filter(self.model_type.id == entity_id).first():  # type: ignore
+        query = self.model_type.__table__.select().filter_by(id=entity_id)  # type: ignore
+
+        if entry := self.db.execute(str(query.compile(dialect=sqlite.dialect())), [entity_id]).fetchone():
+            entry = self.model_type.from_dict(entry)  # type: ignore
             return self.converter._model_to_entity(entry)
         raise ModelNotFound(str(self.model_type.__name__), entity_id)
 
     def _load_all(self, filters: Optional[List[Dict]] = None) -> List[Entity]:
-        query = self.db.query(self.model_type)
+        query = self.model_type.__table__.select()  # type: ignore
         entities: List[Entity] = []
 
         for f in filters or [{}]:
             filtered_query = query.filter_by(**f)
             try:
-                entities.extend([self.converter._model_to_entity(m) for m in filtered_query.all()])
+                entries = self.db.execute(
+                    str(filtered_query.compile(dialect=sqlite.dialect())),
+                    [self.__serialize_filter_values(val) for val in list(f.values())],
+                ).fetchall()
+
+                entities.extend(
+                    [self.converter._model_to_entity(self.model_type.from_dict(m)) for m in entries]  # type: ignore
+                )
             except NoResultFound:
                 continue
         return entities
 
     def _delete(self, entity_id: str):
-        number_of_deleted_entries = self.db.query(self.model_type).filter_by(id=entity_id).delete()
-        if not number_of_deleted_entries:
-            raise ModelNotFound(str(self.model_type.__name__), entity_id)
+        delete_query = self.model_type.__table__.delete().filter_by(id=entity_id)  # type: ignore
+        cursor = self.db.execute(str(delete_query.compile(dialect=sqlite.dialect())), [entity_id])
         self.db.commit()
 
+        if cursor.rowcount == 0:
+            raise ModelNotFound(str(self.model_type.__name__), entity_id)
+
+        if cursor.rowcount == 0:
+            raise ModelNotFound(str(self.model_type.__name__), entity_id)
```

Comment:
The `if` block is duplicated I believe. Also, should we run the `if` before commit the connection?

```suggestion
        if cursor.rowcount == 0:
            raise ModelNotFound(str(self.model_type.__name__), entity_id)
            
        self.db.commit()
```